tests/vppinputasync.cpp

```diff
@@ -91,18 +126,24 @@ bool VppInputAsync::read(SharedPtr<VideoFrame>& frame)
     }
     frame = m_queue.front();
     m_queue.pop_front();
+    if (m_startupSize) {
+        //we make a buffer loop here.
+        m_queue.push_back(frame);
+    }
     m_cond.signal();
     return true;
 }
 
 VppInputAsync::~VppInputAsync()
 {
-    {
-        AutoLock lock(m_lock);
-        m_quit = true;
-        m_cond.signal();
+    if (m_thread != INVALID_THREAD_ID) {
+        {
+            AutoLock lock(m_lock);
+            m_quit = true;
+            m_cond.signal();
+        }
+        pthread_join(m_thread, NULL);
```

Comment:
no, we only join thread when we created it. if m_thread == INVALID_THREAD_ID, we did not create the thread.